controllers/Package.js

```diff
@@ -9,23 +9,31 @@ exports.getOnePackage = (req, res) => {
           message: 'Package with that id was not found ' + req.params.packageId,
         })
       }
-      Category.findById(packagebyid.categories)
+      Category.find({
+        _id: {
+          $in: packagebyid.categories,
+        },
+      })
         .select({
           _id: 0,
         })
-        .then(categorydetails => {
-          Itenary.findById(packagebyid.itenary)
+        .then(categories => {
+          Itenary.find({
+            _id: {
+              $in: packagebyid.itenary,
+            },
+          })
             .select({
               _id: 0,
             })
-            .then(itenarydetails => {
-              if (!itenarydetails || !categorydetails) {
+            .then(itenary => {
+              if (!itenary || !categories) {
```

Comment:
Why this if condition?

Reply:
Is the returning an empty array even if there are no matching categories or itenaries?